llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/VectorExt/IR/VectorExtAttrs.td

```diff
@@ -99,5 +99,170 @@ def LayoutAttr : IREEVectorExt_Attr<"Layout",
   }];
 }
 
-#endif // IREE_DIALECT_VECTOREXT_ATTRS
+def LayoutV2Attr : IREEVectorExt_Attr<"LayoutV2",
+      [ DeclareAttrInterfaceMethods<VectorLayoutInterface> ]> {
+  let mnemonic = "layoutv2";
+  let summary = [{A layout representing how a GPU thread hierarchy is distributed over a shape}];
+  let description = [{
+    This layout explicitly defines how a shape is distributed over a tile
+    heirarchy. We consider the following levels of hierarchy:
+
+    1. Subgroups per Workgroup
+    2. Threads per Subgroup
+    3. Elements per Thread
+
+    Each level of heirarchy represents a tile over the next level of
+    heirarchy, except the last level, which represents elements. The tiling is
+    done on each dimension of the original vector shape. For example, for a
```

Comment:
I would add that each tiling level represents a count of tiles, rather than a list of tile sizes.